src/components/MetricsOptions/__tests__/MetricsOptionsBar.test.tsx

```diff
@@ -11,44 +11,45 @@ const lastOptionsChanged = () => {
 describe('MetricsOptionsBar', () => {
   it('renders initial layout', () => {
     const wrapper = shallow(<MetricsOptionsBar onOptionsChanged={jest.fn()} />);
-    const eltDuration = wrapper.find('#duration');
-    expect(eltDuration.length).toBe(1);
-    const eltStep = wrapper.find('#step');
-    expect(eltStep.length).toBe(1);
-    const eltRateInterval = wrapper.find('#rateInterval');
-    expect(eltRateInterval.length).toBe(1);
+    expect(wrapper).toMatchSnapshot();
```

Comment:
When do you record the snapshot ?
If this is to ensure that only intended changes in the UI are detected fine, but it is an open discussion the granularity of UI tests.

Reply:
Enzyme creates a snapshot when the test is run and store it to a file. If the file exists, snapshots are compared and the test fails if it doesn't match. If the file doesn't exist (which is the case for the first run obviously), the test passes.
So, in this PR it doesn't test anything, but in future PRs it will.

It looks like a convenient way to test UI without giving headaches.

When we want to change the UI, we just delete the snapshot, run the test and it will pass. We must just make sure the diff with the old snapshot is consistent with the intended result. And reviewers will check that, too.

I think we should give it a try. It seems to be a very common pattern with testing with enzyme. @aljesusg uses it as well in his PR https://github.com/kiali/swsui/pull/100